test/OpenTelemetry.AutoInstrumentation.Tests/OtelLoggingTests.cs

```diff
@@ -32,94 +37,59 @@ public void WhenNoFileSizeIsConfigured_Then_DefaultIsUsed()
-            OtelLogging.GetConfiguredFileSizeLimitBytes().Should().Be(10 * 1024 * 1024);
-        }
-        finally
-        {
-            Environment.SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOG_FILE_SIZE", read);
-        }
+        Environment.SetEnvironmentVariable("OTEL_DOTNET_AUTO_LOG_FILE_SIZE", "-1");
+
+        OtelLogging.GetConfiguredFileSizeLimitBytes().Should().Be(10 * 1024 * 1024);
     }
 
     [Fact]
     public void WhenLogLevelIsNotConfigured_Then_DefaultIsUsed()
     {
-        var read = Environment.GetEnvironmentVariable("OTEL_LOG_LEVEL");
-        try
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", string.Empty);
-
-            OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Information);
-        }
-        finally
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", read);
-        }
+        Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", null);
+
+        OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Information);
     }
 
     [Fact]
     public void WhenInvalidLogLevelIsConfigured_Then_DefaultIsUsed()
     {
-        var read = Environment.GetEnvironmentVariable("OTEL_LOG_LEVEL");
-        try
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "invalid");
-
-            OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Information);
-        }
-        finally
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", read);
-        }
+        Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "invalid");
+
+        OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Information);
     }
 
     [Fact]
     public void WhenValidLogLevelIsConfigured_Then_ItIsUsed()
     {
-        var read = Environment.GetEnvironmentVariable("OTEL_LOG_LEVEL");
-        try
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "warn");
-
-            OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Warning);
-        }
-        finally
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", read);
-        }
+        Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "warn");
+
+        OtelLogging.GetConfiguredLogLevel().Should().Be(LogLevel.Warning);
     }
 
     [Fact]
     public void WhenNoLoggingIsConfigured_Then_LogLevelHasNoValue()
     {
-        var read = Environment.GetEnvironmentVariable("OTEL_LOG_LEVEL");
-        try
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "none");
-
-            OtelLogging.GetConfiguredLogLevel().Should().BeNull();
-        }
-        finally
-        {
-            Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", read);
-        }
+        Environment.SetEnvironmentVariable("OTEL_LOG_LEVEL", "none");
+
+        OtelLogging.GetConfiguredLogLevel().Should().BeNull();
+    }
+
+    public void Dispose()
+    {
+        UnsetLogLevelEnvVar();
+    }
```

Comment:
nit: move it after constructor for better readability

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2277/commits/c78e1f0b869a3cb19d1070bfb2d4d963b416bc0a